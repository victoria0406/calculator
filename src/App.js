import './App.css';
import { useState } from 'react';
import {Provider, useDispatch, useSelector} from 'react-redux';
import reducer, {add_input, calculate, save} from './reducer';
import { createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

const AppWrapper = () => {
  const store = createStore(reducer, composeWithDevTools()); 

  return (
    <Provider store={store}> 
      <App />
    </Provider>
  )
}

function App() {
  const {input_list, answer} =  useSelector(state => ({
    input_list:state.input_list,
    answer:state.answer
  }));
  const dispatch = useDispatch();
  return (
    <div>
      <div>테스트용 계산기</div>
      <div>{input_list}</div>
      <div>{answer}</div>
      <div>
        {[...Array(10).keys()].map((ele)=>(
          <button key={ele} onClick={()=>{dispatch(add_input(String(ele)));dispatch(calculate());}}>{ele}</button>
        ))
        }
      </div>
      <div>
        <button onClick={()=>{dispatch(add_input("("))}}>{"("}</button>
        <button onClick={()=>{dispatch(add_input(")"));dispatch(calculate());}}>{")"}</button>
        <button onClick={()=>{dispatch(add_input("+"))}}>+</button>
        <button onClick={()=>{dispatch(add_input("-"))}}>-</button>
        <button onClick={()=>{dispatch(add_input("x"))}}>x</button>
        <button onClick={()=>{dispatch(add_input("/"))}}>/</button>
        <button onClick={()=>{dispatch(save())}}>=</button>
      </div>
      <button>=</button>
    </div>
  );
}

export default AppWrapper;
