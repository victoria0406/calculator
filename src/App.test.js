import React from "react";
import { render } from "@testing-library/react";
import App from "./App";

describe('<App/>', ()=>{
    //DOM되는 과정에서 문제가 없는가?
    it("match snapshot", ()=>{
        const utils = render(<App/>);
        expect(utils.container).toMatchSnapshot();
    })
    //지정한 component들이 DOM과정에서 다 드러났는가?
    it('shows the props correctly',()=>{
        const utils = render(<App/>);
        utils.getByText(/[0-9]/); //근데 이것
    })

})