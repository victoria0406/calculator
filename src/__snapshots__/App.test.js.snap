// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> match snapshot 1`] = `
<div>
  <div>
    <div>
      테스트용 계산기
    </div>
    <div />
    <div>
      0
    </div>
    <div>
      <button>
        0
      </button>
      <button>
        1
      </button>
      <button>
        2
      </button>
      <button>
        3
      </button>
      <button>
        4
      </button>
      <button>
        5
      </button>
      <button>
        6
      </button>
      <button>
        7
      </button>
      <button>
        8
      </button>
      <button>
        9
      </button>
    </div>
    <div>
      <button>
        (
      </button>
      <button>
        )
      </button>
      <button>
        +
      </button>
      <button>
        -
      </button>
      <button>
        x
      </button>
      <button>
        /
      </button>
      <button>
        =
      </button>
    </div>
    <button>
      =
    </button>
  </div>
</div>
`;
